Kops cleanup gives error

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**What happened**:
When running the e2e tests, the trap attempts to run [kops-finish](https://github.com/kubernetes/cloud-provider-aws/blob/master/hack/e2e/run.sh#L130):

```
trap kops-finish EXIT
```

However, the `${test-run}` directory seems to be missing when its run, so perhaps cleanup happens before trap can execute?

```
 Ginkgo ran 1 suite in 2m57.717745585s
Test Suite Passed
/home/prow/go/src/k8s.io/cloud-provider-aws
./hack/e2e/run.sh: line 125: /home/prow/go/src/k8s.io/cloud-provider-aws/_output/test/20220819152419/kops: No such file or directory
make: *** [Makefile:133: test-e2e] Error 127
+ EXIT_VALUE=2
+ set +o xtrace
Cleaning up after docker in docker. 
```

See https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-aws/443/pull-cloud-provider-aws-e2e/1560647514955190272

**What you expected to happen**:
Cleanup to happen successfully.

**How to reproduce it (as minimally and precisely as possible)**:
Run the e2e tests in prow by opening a PR.  

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

--- a/hack/e2e/run.sh
+++ b/hack/e2e/run.sh
@@ -121,14 +121,7 @@ aws ecr get-login-password --region "${REGION}" | docker login --username AWS --
 docker tag "${BUILD_IMAGE}" "${IMAGE_NAME}:${IMAGE_TAG}"
 docker push "${IMAGE_NAME}:${IMAGE_TAG}"
 
-function kops-finish {
-    "${test_run}"/kops delete cluster --name "${CLUSTER_NAME}" --yes
-}
-
 if [[ "${UP}" = "yes" ]]; then
-
-    trap kops-finish EXIT
-
     kubetest2 kops \
       -v 2 \
       --up \
@@ -151,3 +144,7 @@ fi
 pushd ./tests/e2e
 ginkgo . -p -nodes="${GINKGO_NODES}" -v --focus="${GINKGO_FOCUS}" --skip="${GINKGO_SKIP}" "" -- -kubeconfig="${KUBECONFIG}" -report-dir="${test_run}" -gce-zone="${ZONES%,*}" "${EXPANDED_TEST_EXTRA_FLAGS}"
 popd
+
+if [[ "${DOWN}" = "yes" ]]; then
+    ${test_run}/kops delete cluster --name "${CLUSTER_NAME}" --yes
+fi

